plugin/tower/pkg/controller/policy/controller.go

```diff
@@ -0,0 +1,820 @@
+		if err != nil && !errors.IsNotFound(err) {
+			return fmt.Errorf("delete policy %s: %s", policyKey, err)
+		}
+	}
+
+	return nil
+}
+
+// parseIsolationPolicy convert schema.SecurityPolicy to []v1alpha1.SecurityPolicy
+func (c *Controller) parseSecurityPolicy(securityPolicy *schema.SecurityPolicy) ([]v1alpha1.SecurityPolicy, error) {
+	var policyList []v1alpha1.SecurityPolicy
+
+	applyToPeers, err := c.parseSecurityPolicyApplys(securityPolicy.ApplyTo)
+	if err != nil {
+		return nil, err
+	}
+
+	ingress, egress, err := c.parseNetworkPolicyRules(securityPolicy.Ingress, securityPolicy.Egress)
+	if err != nil {
+		return nil, err
+	}
+
+	policy := v1alpha1.SecurityPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      SecurityPolicyPrefix + securityPolicy.GetID(),
+			Namespace: c.namespace,
+		},
+		Spec: v1alpha1.SecurityPolicySpec{
+			Tier:          constants.Tier1,
+			SymmetricMode: true,
+			AppliedTo:     applyToPeers,
+			IngressRules:  ingress,
+			EgressRules:   egress,
+			PolicyTypes:   []networkingv1.PolicyType{networkingv1.PolicyTypeIngress, networkingv1.PolicyTypeEgress},
+		},
+	}
+	policyList = append(policyList, policy)
+
+	for item := range securityPolicy.ApplyTo {
+		if !securityPolicy.ApplyTo[item].Communicable {
+			continue
+		}
+		// generate intra group policy
+		policy, err := c.generateIntragroupPolicy(securityPolicy.GetID(), &securityPolicy.ApplyTo[item])
+		if err != nil {
+			return nil, err
+		}
+		policyList = append(policyList, *policy)
+	}
+
+	return policyList, nil
+}
+
+// parseIsolationPolicy convert schema.IsolationPolicy to []v1alpha1.SecurityPolicy
+func (c *Controller) parseIsolationPolicy(isolationPolicy *schema.IsolationPolicy) ([]v1alpha1.SecurityPolicy, error) {
+	applyToPeers, err := c.vmAsAppliedTo(isolationPolicy.VM.ID)
+	if err != nil {
+		return nil, err
+	}
+	if len(applyToPeers) == 0 {
+		return nil, nil
+	}
+
+	policy := v1alpha1.SecurityPolicy{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      IsolationPolicyPrefix + isolationPolicy.GetID(),
+			Namespace: c.namespace,
+		},
+		Spec: v1alpha1.SecurityPolicySpec{
+			Tier:          constants.Tier0,
+			SymmetricMode: true,
+			AppliedTo:     applyToPeers,
+			PolicyTypes:   []networkingv1.PolicyType{networkingv1.PolicyTypeIngress, networkingv1.PolicyTypeEgress},
+		},
+	}
+
+	switch isolationPolicy.Mode {
+	case schema.IsolationModeAll:
+		// IsolationModeAll should not create ingress or egress rule
+	case schema.IsolationModePartial:
+		ingress, egress, err := c.parseNetworkPolicyRules(isolationPolicy.Ingress, isolationPolicy.Egress)
+		if err != nil {
+			return nil, err
+		}
+		policy.Spec.IngressRules = ingress
+		policy.Spec.EgressRules = egress
+	}
+
+	return []v1alpha1.SecurityPolicy{policy}, nil
+}
+
+func (c *Controller) generateIntragroupPolicy(securityPolicyID string, appliedPeer *schema.SecurityPolicyApply) (*v1alpha1.SecurityPolicy, error) {
```

Comment:
If a "label" in the ApplyTo only contains 1 VM, do we need any specific process about the generate IntragroupPolicy?

Reply:
The vm relate labels are variable, we should just generate IntragroupPolicy, and don't care about the relation.